packages/next/server/node-web-streams-helper.ts

```diff
@@ -149,19 +149,44 @@ export function renderToInitialStream({
   return ReactDOMServer.renderToReadableStream(element, streamOptions)
 }
 
-export function createHeadInjectionTransformStream(
-  inject: () => Promise<string>
+function createHeadInsertionTransformStream(
+  insert: () => Promise<string>
 ): TransformStream<Uint8Array, Uint8Array> {
-  let injected = false
+  let inserted = false
+  let freezing = false
+  const queueTask =
+    process.env.NEXT_RUNTIME === 'edge' ? globalThis.setTimeout : setImmediate
   return new TransformStream({
     async transform(chunk, controller) {
```

Comment:
Is there some kind of fast path we can check in case there are no callbacks queued so we can bail early?